src/sql/workbench/browser/designer/designer.ts

```diff
@@ -836,6 +824,53 @@ export class Designer extends Disposable implements IThemable {
 					headerRowHeight: TableHeaderRowHeight,
 					editorLock: new Slick.EditorLock()
 				});
+
+				let moveRowsPlugin = new RowMoveManager({
+					cancelEditOnDrag: true,
+					id: 'moveRow',
+					iconCssClass: Codicon.grabber.classNames,
+					title: localize('designer.moveRowText', 'Move Row'),
+					width: 20,
+					resizable: false,
+					isFontIcon: true,
+					behavior: 'selectAndMove'
+				});
+				moveRowsPlugin.onBeforeMoveRows.subscribe((e, data) => {
+					let test = data;
+				});
+				moveRowsPlugin.onMoveRows.subscribe((e, args) => {
+					let test = args;
+				});
+
+				table.grid.registerPlugin(moveRowsPlugin);
+
+
+
+				table.grid.onDragStart.subscribe((e, dd) => {
+					let test = dd;
+				});
+
+				table.grid.onDrag.subscribe(function (e, dd) {
+					let test = dd;
+				});
+
+				table.grid.onAddNewRow.subscribe((e, args) => {
+					let item = { name: 'New task', complete: false };
+					jQuery.extend(item, args.item);
+					// data.push(item); // data is probably the grid data already there
+					// table.grid.invalidateRows([data.length - 1]);
+					table.grid.updateRowCount();
+					table.grid.render();
+				});
+
+				this._register(table.onContextMenu((e) => {
+					let edit: DesignerEdit = {
```

Comment:
this should be created in the openContextMenu() method, if you need you can pass in the view and path.